cmd/cert-checker/main.go

```diff
@@ -270,11 +270,7 @@ func (c *certChecker) checkCert(cert core.Certificate) (problems []string) {
 			id := core.AcmeIdentifier{Type: core.IdentifierDNS, Value: name}
 			// TODO(https://github.com/letsencrypt/boulder/issues/3371): Distinguish
 			// between certificates issued by v1 and v2 API.
-			checkFunc := c.pa.WillingToIssue
-			if features.Enabled(features.WildcardDomains) {
-				checkFunc = c.pa.WillingToIssueWildcard
-			}
-			if err = checkFunc(id); err != nil {
+			if err = c.pa.WillingToIssueWildcard(id); err != nil {
```

Comment:
Makes sense, :+1:

Reply:
Actually scratch that, we still use `WillingToIssue` in the v1 API since it doesn't support wildcards, and as such I think the distinction does still make sense.

Reply:
Agreed on the distinction still being important while we maintain the V1 API.